pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -111,7 +111,8 @@
     @Override
     @NotNull
     public CompletableFuture<SchemaVersion> putSchemaIfAbsent(String schemaId, SchemaData schema,
-                                                              SchemaCompatibilityStrategy strategy) {
+                                                              SchemaCompatibilityStrategy strategy,
+                                                              boolean isAllowCreateSchema) {
```

Comment:
Looks weird here, i think we'd better add a method to get schema version by schemaData which producer carried, if the the exists schemas contains the schema that producer carried, producer can connect success, otherwise if not allow schema auto update, reject the producer.